tools/addScopes/addScopes.cpp

```diff
@@ -51,10 +52,48 @@ int main(int argc, char* argv[])
         if(secmgr == NULL)
         {
             printf("Security manager can't be created\n");
+            releaseAtoms();
             return -1;
         }
         bool ok = secmgr->createUserScopes();
         printf(ok ? "User scopes added\n" : "Some scopes not added\n");
+
+        //Clear permission caches?
+        if (argc == 3 && 0==stricmp(argv[2], "-c"))
+        {
+            //Clear ESP Cache
+            StringBuffer sysuser;
+            StringBuffer passbuf;
+            seccfg->getProp(".//@systemUser", sysuser);
+            seccfg->getProp(".//@systemPassword", passbuf);
+
+            if (0 == sysuser.length())
+            {
+                printf("Error in configuration file %s - systemUser not specified", argv[1]);
+                releaseAtoms();
+                return -1;
+            }
+
+            if (0 == passbuf.length())
+            {
+                printf("Error in configuration file %s - systemPassword not specified", argv[1]);
+                releaseAtoms();
+                return -1;
+            }
+            Owned<ISecUser> user = secmgr->createUser(sysuser.str());
+            ISecCredentials& cred = user->credentials();
+            StringBuffer decPwd;
+            decrypt(decPwd, passbuf.str());
+            cred.setPassword(decPwd.str());
+            ok = secmgr->clearPermissionsCache(*user);
+            printf(ok ? "ESP Cache cleared\n" : "Error clearing ESP Cache\n");
```

Comment:
Actually, there is no reason to clear the ESP cache, since it does not manage file scopes.  Same with  ECLServer, since it also calls to Dali for file access. I will remove the call to the ESP